### PR TITLE
feature (ref 34969): add missed endif in twig

### PR DESCRIPTION
**Pull request by jalalifardemos**
**Ticket:** https://yaits.demos-deutschland.de/T34969

Description: add missed endif in statement version twig

### How to review/test
code review or browser

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig [hunk 1]
```diff
@@ -26,9 +26,11 @@
 
         {% if draftStatementVersions is defined and  draftStatementVersions|length > 0 %}
 
-            {% for index, statement in draftStatementVersions if index < lastIndex %}
-                <h2>{{ "date"|trans }}: {{ statement.versionDate|dplanDate('d.m.Y | H:i') }}</h2>
-                {{ statement.text|wysiwyg }}
+            {% for index, statement in draftStatementVersions%}
+                {% if index < lastIndex  %}
```
Comment (mussbach) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:30]: you can do this more concise with the twig filter like for example in c214fe94

**Review by mussbach — CHANGES_REQUESTED** [review]

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig [hunk 2]
```diff
@@ -26,9 +26,11 @@
...
+                    <h2>{{ "date"|trans }}: {{ statement.versionDate|dplanDate('d.m.Y | H:i') }}</h2>
+                    {{ statement.text|wysiwyg }}
+                {% endif %}
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:33]: why is this change neccesssary? 
The Doc tells me that it should work the way it was before https://twig.symfony.com/doc/2.x/tags/for.html#adding-a-condition
Could it be, that setting some brackets help to solve the Bug?

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig (hunk 2 above)
Comment (jalalifardemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:33]: It gives error with this format

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig (hunk 1 above)
Comment (jalalifardemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:30]: Can you please send a link? Token is not clickable.

Comment (mussbach) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:30]: Yes, sorry #2033

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig (hunk 2 above)
Comment (mussbach) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:33]: We use Twig 3

Comment (salisdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:33]: :bulb:  :-/

Comment (salisdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:33]: then, what @mussbach says (filter)

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig (hunk 1 above)
Comment (jalalifardemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig]: done in [5f79a5f](https://github.com/demos-europe/demosplan-core/pull/2152/commits/5f79a5f388775759f295fe310d31312f4b32d7bd)

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig (hunk 2 above)
Comment (jalalifardemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig]: done in [5f79a5f](https://github.com/demos-europe/demosplan-core/pull/2152/commits/5f79a5f388775759f295fe310d31312f4b32d7bd)

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig [hunk 3]
```diff
@@ -26,9 +26,9 @@
 
         {% if draftStatementVersions is defined and  draftStatementVersions|length > 0 %}
 
-            {% for index, statement in draftStatementVersions if index < lastIndex %}
-                <h2>{{ "date"|trans }}: {{ statement.versionDate|dplanDate('d.m.Y | H:i') }}</h2>
-                {{ statement.text|wysiwyg }}
+            {% for statement in draftStatementVersions|filter((statement, index) => index < lastIndex)%}
+                    <h2>{{ "date"|trans }}: {{ statement.versionDate|dplanDate('d.m.Y | H:i') }}</h2>
+                    {{ statement.text|wysiwyg }}
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:31]: nitpicks
```suggestion
            {% for statement in draftStatementVersions|filter((_statement, index) => index < lastIndex) %}
                <h2>{{ "date"|trans }}: {{ statement.versionDate|dplanDate('d.m.Y | H:i') }}</h2>
                {{ statement.text|wysiwyg }}
```

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig (hunk 3 above)
Comment (jalalifardemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig]: done [e90bb93](https://github.com/demos-europe/demosplan-core/pull/2152/commits/e90bb93d485bba68f55447809b380bfed36cd268)

**Review by salisdemos — APPROVED** [review]

File: templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig [hunk 4]
```diff
@@ -26,7 +26,7 @@
 
         {% if draftStatementVersions is defined and  draftStatementVersions|length > 0 %}
 
-            {% for index, statement in draftStatementVersions if index < lastIndex %}
+            {% for statement in draftStatementVersions|filter((_statement, index) => index < lastIndex) %}
```
Comment (mussbach) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:29]: Did you try it? How is `lastIndex` passed to the array function?

Comment (mussbach) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanStatement/versions_of_statement.html.twig:29]: Thank you, did not know that

**Review by mussbach — APPROVED** [review]
lgtm, code review only

**Pull request merged**